kensu/utils/kensu.py

```diff
@@ -335,6 +335,20 @@ def in_mem(self, var_name):
     def name_for_stats(self, input_name):
         return input_name.replace('.', '_')
 
+    def extract_stats(self, stats_df):
+        try:
+            stats = self.extractors.extract_stats(stats_df)
+        except Exception as e:
+            logging.info(f'stats extraction from {type(stats_df)} failed', e)
+            if isinstance(stats_df, dict):
+                # FIXME: this is weird logic here... we'd post the actual data instead of stats!
+                stats = stats_df
```

Comment:
didn't change logic here yet.
@SammyEK why do we have this? seem something quite buggy to me... (?)

Reply:
in that case I suppose it's because the stats are already computed, as it is a dict

Reply:
then, in future, we should change this sort of  `data transfer object` which now is `dict` into some more specific class like `KensuStatsContainer` (or existing `KensuDatasourceAndSchema` class) - more understandable & less error prone